scipy/stats/_stats.pyx

```diff
@@ -753,3 +753,90 @@ def gaussian_kernel_estimate(points, values, xi, precision, dtype, real _=0):
                 estimate[j, k] += values_[i, k] * arg
 
     return np.asarray(estimate)
+
+
+@cython.wraparound(False)
+@cython.boundscheck(False)
+def gaussian_kernel_estimate_log(points, values, xi, precision, dtype, real _=0):
+    """
+    def gaussian_kernel_estimate_log(points, real[:, :] values, xi, precision)
+
+    Evaluate the log of the estimated pdf on a provided set of points.
+
+    Parameters
+    ----------
+    points : array_like with shape (n, d)
+        Data points to estimate from in ``d`` dimensions.
+    values : real[:, :] with shape (n, p)
+        Multivariate values associated with the data points.
+    xi : array_like with shape (m, d)
+        Coordinates to evaluate the estimate at in ``d`` dimensions.
+    precision : array_like with shape (d, d)
+        Precision matrix for the Gaussian kernel.
+
+    Returns
+    -------
+    estimate : double[:, :] with shape (m, p)
+        The log of the multivariate Gaussian kernel estimate evaluated at the
+        input coordinates.
+    """
+    cdef:
+        real[:, :] points_, xi_, values_, log_values_
+        real[:, :] whitening, estimate, estimate_, zeros_estimate_
+        real[:] max_, zeros_max
+        int i, j, k
+        int n, d, m, p
+        real arg, residual, log_arg, log_norm
+
+    n = points.shape[0]
+    d = points.shape[1]
+    m = xi.shape[0]
+    p = values.shape[1]
+
+    # Rescale the data
+    whitening = np.linalg.cholesky(precision).astype(dtype, copy=False)
+    points_ = np.dot(points, whitening).astype(dtype, copy=False)
+    xi_ = np.dot(xi, whitening).astype(dtype, copy=False)
+    values_ = values.astype(dtype, copy=False)
+
+    # Log transformation of values_. It is faster than np.log(values_).
+    log_values_ = np.empty((n, p), dtype)
+    for i in range(n):
+        for k in range(p):
+            log_values_[i, k] = math.log(values_[i, k])
+
+    # Evaluate the normalisation
+    log_norm = (- d / 2) * math.log(2 * PI)
+    for i in range(d):
+        log_norm += math.log(whitening[i, i])
+
+    # Create the result array and evaluate the weighted sum
+    estimate = np.zeros((m, p), dtype)
+    zeros_max = np.zeros((p,), dtype) - np.inf
+    empty_estimate_ = np.empty((p, n), dtype)
+    for j in range(m):
```

Comment:
The loop variables `i` and `j` are exactly reversed compared to `gaussian_kernel_estimate`, can you please make them the same?

Reply:
If I corrected the order of `i` and `j`, it would not work well because
```python
# Calculate logsumexp
for k in range(p):
    for i in range(n):
        estimate[j, k] += math.exp(estimate_[k, i] - max_[k])
    estimate[j, k] = math.log(estimate[j, k])
    estimate[j, k] += max_[k]
```
this block depends on `j`. Currently, the loop logic is complicated so it's hard to make it align with `gaussian_kernel_estimate`. The code would be simple if we could separate `logsumexp` from this loop.

Reply:
I made them the same.